### PR TITLE
Fixed #30656 -- Added QuerySet.bulk_update() to the database optimization docs.

### PR DESCRIPTION
**Pull request by MisterRios**
Hi,

I ended up restructuring the `Insert in bulk` section because it didn't make sense to me.
Inserting and creating were lumped together with the header `Insert in bulk`, so it took me a bit to realize what the `my_band` section was doing there.

The Update in bulk section is the new bit, and the part that directly addresses the ticket.

### CODE REVIEW

File: docs/topics/db/optimization.txt [hunk 1]
```diff
@@ -373,3 +413,8 @@ This also applies to :class:`ManyToManyFields
     my_band.members.add(my_friend)
 
 ...where ``Bands`` and ``Artists`` have a many-to-many relationship.
+
+List unpacking also works to add querysets or lists of objects::
+
+    new_band_members = [me, my_friend]
+    myband.members.add(*new_band_members)
```
Comment (felixxm) [inline docs/topics/db/optimization.txt]: I don't see much value in these change and it's definitely not related with this ticket. Please remove it.

File: docs/topics/db/optimization.txt [hunk 2]
```diff
@@ -341,8 +341,13 @@ it on a ``QuerySet`` by calling
 
 Adding an index to your database may help to improve ordering performance.
 
-Insert in bulk
-==============
+Use bulk methods
+================
+
+Built-in bulk methods can reduce multiple queries to a single query.
```
Comment (felixxm) [inline docs/topics/db/optimization.txt]: I would say:
```
Use bulk methods to reduce the number of SQL statements.
```

**Comment by felixxm** [comment]
Please swap commits: 1st - refactoring, 2nd - adding a `bulk_update()`.

**Comment by MisterRios** [comment]
@felixxm Wow! Much nicer that way. Thanks for the feedback!

**Comment by felixxm** [comment]
@MisterRios Thanks

**Pull request merged**